src/spec/source.cr

```diff
@@ -8,8 +8,7 @@ module Spec
   def self.read_line(file, line)
     return nil unless File.file?(file)
 
-    lines = lines_cache[file] ||= File.read_lines(file)
-    lines[line - 1]?
+    lines_cache.put_if_absent(file) { File.read_lines(file) }[line - 1]?
```

Comment:
Local variable makes it more readable

Reply:
yup :)

Reply:
Once could argue that the `.put_if_absent(file) { File.read_lines(file) } << item` constructs are bit overly complicated as well.